File: benchmark/c/cpw/sum_factors_number_1/sum_factors_number_1.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int sum_factors_number_1 ( int n ) ;
int sum_factors_number_1 ( int n ) {
  int res = 1;
  for ( int i = 2;
  i <= sqrt ( n );
  i ++ ) {
    int curr_sum = 1;
    int curr_term = 1;
    while ( n % i == 0 ) {
      n = n / i;
      curr_term *= i;
      curr_sum += curr_term;
    }
    res *= curr_sum;
  }
  if ( n >= 2 ) res *= ( 1 + n );
  return res;
}

File: experimental data/RQ2/translation_result/success/c/cpw/gpt4/BaseRepair/few-shot-False/cot-ast/candidates-1/sum_factors_number_1/sum_factors_number_1.rs
fn sum_factors_number_1(n: i32) -> i32 {
    let mut res: i32 = 1;
    let mut n_remaining = n;
    let sqrt_n = (n_remaining as f64).sqrt() as i32;

    let mut i: i32 = 2;
    while i <= sqrt_n {
        if n_remaining % i == 0 {
            let mut curr_sum: i32 = 1;
            let mut curr_term: i32 = 1;
            while n_remaining % i == 0 {
                n_remaining = n_remaining.wrapping_div(i);
                curr_term = curr_term.wrapping_mul(i);
                curr_sum = curr_sum.wrapping_add(curr_term);
            }
            res = res.wrapping_mul(curr_sum);
        }
        i = i.wrapping_add(1);
    }

    if n_remaining >= 2 {
        res = res.wrapping_mul(1i32.wrapping_add(n_remaining));
    }

    res
}